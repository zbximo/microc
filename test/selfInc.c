void main()
{
    Int a;
    Int b;
    a = 1;
    b = a++ * 3 * a;
    print a;
    print b;
    // Int a;
    // Int b;
    // a = 1;
    // b = a++;
    // print a;
    // print b;
    // b = a--;
    // print a;
    // print b;
    // b = --a;
    // print a;
    // print b;
    // b = ++a;
    // print a;   
    // b = 2;
    // print 2*b--; 
    // b = 2;
    // print --b;

}