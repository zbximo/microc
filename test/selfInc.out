24 19 0 5 25 15 1 15 1 13 0 1 12 15 -1 13 0 1 1 13 13 11 0 1 1 12 12 15 -1 13 11 22 15 -1 13 0 1 1 11 22 15 -1 13 0 1 1 13 13 11 0 1 2 12 12 15 -1 13 11 22 15 -1 13 0 1 1 11 22 15 -1 13 0 1 1 0 1 2 13 11 12 15 -1 13 11 22 15 -1 13 0 1 1 11 22 15 -1 13 0 1 1 0 1 1 13 11 12 15 -1 13 11 22 15 -1 13 0 1 1 0 2 12 15 -1 0 2 13 0 1 1 13 0 1 1 11 0 1 2 12 3 22 15 -1 13 0 1 1 0 2 12 15 -1 0 1 2 13 0 1 1 11 22 21 2